### ISSUE TITLE
LAMB and amp

### ISSUE THREAD

**Issue by sgunasekar** [facebookresearch/deit#182]
I recently used LAMB (from timm library) for the first time to train resmlp and noticed that I kept getting nan-losses when I was using mixed precision (AMP support from torch) Based on discussion in https://github.com/facebookresearch/deit/issues/106#issuecomment-915187091 I dropped AMP and everything works fine now.  Training resmlp with adamw without mixed precision but adding gradient clipping also works without leading to loss (albeit poorer performance). Tl;Dr ADAMW+AMP, LAMB+noAMP works but LAMB+AMP creates nan-losses. 

I am wondering if there is some fundamental issue with using LAMB and AMP -- since when using AMP, we need to unscale the gradients before clipping?

**Comment by TouvronHugo** [comment]
Hi @sgunasekar,
Yes, there is a version of Lamb adapted for mixed precision training. 
Indeed, if you use gradient clipping before clipping gradient it is necessary to unscale the gradients. 
Best,
Hugo
